### DISCUSSION TITLE
Uploading in bulk to storage with a python script

### DISCUSSION THREAD

**Discussion by ethanfinkel** [supabase/supabase#9963] [Questions] [answered]
There's a bunch of files I want to add to the storage bucket and associate them with some rows in my db. I have neat little python script to get all the files I want via some web scraping, and I'd like to upload them to the bucket in bulk via Python. All the documentation I've found has this done via the supabase client in JS.

Is there a clean way for me to upload these files to storage straight from my python script?

**Comment by GaryAustin1** [comment, marked as answer]
You can call the Storage REST API directly:
https://supabase.github.io/storage-api/#/object/post_object__bucketName___wildcard_

There is also a python client (I do not know its status).  Either using it, or looking at it's code might be helpful.
https://github.com/supabase-community/storage-py

**Reply by ethanfinkel** [reply]
Thanks Gary that storage rest API worked perfectly
